File: java/Test7.java
public class Test7 {
    public static void main(String[] args) {
        Tree[] t = new TestCase().t;
        System.out.println("[subst]");
        for (int i = 0; i < t.length; i += 2) {
            System.out.print((i/2) + ": ");
            System.out.println(t[i].subst(3, t[i+1]));
        }
    }
}

File: ocaml/test7.ml
open Tree
open Testcase

let () =
  let rec separate l =
    match l with
      [] -> []
    | x::[] -> []
    | x::y::tl -> (x, 3, y)::(separate tl) in
  let l = separate test_case in
  print_endline "[subst]";
  print_list (fun t -> (string_of_tree (subst t))) l 0
